File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/NumberConverter.cpp
#include <string>

class NumberConverter {
public:
    static std::string decimal_to_binary(long long decimal_num) {
        if (decimal_num == 0) return "0";
        bool neg = decimal_num < 0;
        unsigned long long val = neg ? -static_cast<unsigned long long>(decimal_num) : decimal_num;
        std::string res;
        while (val > 0) {
            res = static_cast<char>('0' + val % 2) + res;
            val /= 2;
        }
        if (neg) res = "-" + res;
        return res;
    }

    static long long binary_to_decimal(const std::string& binary_num) {
        return std::stoll(binary_num, nullptr, 2);
    }

    static std::string decimal_to_octal(long long decimal_num) {
        if (decimal_num == 0) return "0";
        bool neg = decimal_num < 0;
        unsigned long long val = neg ? -static_cast<unsigned long long>(decimal_num) : decimal_num;
        std::string res;
        while (val > 0) {
            res = static_cast<char>('0' + val % 8) + res;
            val /= 8;
        }
        if (neg) res = "-" + res;
        return res;
    }

    static long long octal_to_decimal(const std::string& octal_num) {
        return std::stoll(octal_num, nullptr, 8);
    }

    static std::string decimal_to_hex(long long decimal_num) {
        if (decimal_num == 0) return "0";
        bool neg = decimal_num < 0;
        unsigned long long val = neg ? -static_cast<unsigned long long>(decimal_num) : decimal_num;
        std::string res;
        while (val > 0) {
            int rem = val % 16;
            if (rem < 10) res = static_cast<char>('0' + rem) + res;
            else res = static_cast<char>('a' + rem - 10) + res;
            val /= 16;
        }
        if (neg) res = "-" + res;
        return res;
    }

    static long long hex_to_decimal(const std::string& hex_num) {
        return std::stoll(hex_num, nullptr, 16);
    }
};

File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/NumberConverter.py
class NumberConverter:
    @staticmethod
    def decimalToBinary(decimalNum: int) -> str:
        # Simulate Java's Integer.toBinaryString: 32-bit two's complement for negatives
        return bin(decimalNum & 0xFFFFFFFF)[2:]

    @staticmethod
    def binaryToDecimal(binaryNum: str) -> int:
        # Mimic Integer.parseInt with radix 2, including range checks for 32-bit int
        val = int(binaryNum, 2)
        if val < -2**31 or val > 2**31 - 1:
            raise ValueError("out of int range")
        return val

    @staticmethod
    def decimalToOctal(decimalNum: int) -> str:
        # Java's Integer.toOctalString: 32-bit unsigned representation for negatives
        return oct(decimalNum & 0xFFFFFFFF)[2:]

    @staticmethod
    def octalToDecimal(octalNum: str) -> int:
        val = int(octalNum, 8)
        if val < -2**31 or val > 2**31 - 1:
            raise ValueError("out of int range")
        return val

    @staticmethod
    def decimalToHex(decimalNum: int) -> str:
        # Java's Integer.toHexString: 32-bit unsigned representation for negatives
        return hex(decimalNum & 0xFFFFFFFF)[2:]

    @staticmethod
    def hexToDecimal(hexNum: str) -> int:
        val = int(hexNum, 16)
        if val < -2**31 or val > 2**31 - 1:
            raise ValueError("out of int range")
        return val